backend/routes/user.routes.ts

```diff
@@ -0,0 +1,59 @@
+import express from "express";
+const router = express.Router();
+const passport = require("passport");
+const bcrypt = require("bcrypt");
+const User = require("../models/User");
+
+import { DatabaseUserInterface } from "../utils/Interfaces";
+
+router.post("/signup", async (req, res) => {
+	const { username, password } = req?.body;
+	if (
+		!username ||
+		!password ||
+		typeof username !== "string" ||
+		typeof password !== "string"
+	) {
+		res.send("Improper Values");
+		return;
+	}
+	User.findOne(
+		{ username },
+		async (err: Error, userExist: DatabaseUserInterface) => {
+			if (err) throw err;
+			if (userExist) res.send("User Already Exists");
+			if (!userExist) {
+				const hashedPassword = await bcrypt.hash(password, 10);
+				const newUser = new User({
+					username,
+					password: hashedPassword,
+				});
+				await newUser.save();
+				res.send("success");
```

Comment:
Always send the status code as well. That is a good practice.
You can easily google what status code you should use.
E.g.
404 = page not found
201 = success

'users already exists', maybe 409? Could be. You decide :D